README.rst

```diff
@@ -0,0 +1,179 @@
+.. image:: https://coveralls.io/repos/github/mosquito/aiormq/badge.svg?branch=master
+   :target: https://coveralls.io/github/mosquito/async-class?branch=master
+   :alt: Coveralls
+
+.. image:: https://img.shields.io/pypi/l/async-class
+   :target: https://pypi.org/project/async-class
+   :alt: License
+
+.. image:: https://github.com/mosquito/async-class/workflows/tests/badge.svg
+   :target: https://github.com/mosquito/async-class/actions?query=workflow%3Atests
+   :alt: Build status
+
+.. image:: https://img.shields.io/pypi/wheel/async-class
+   :target: https://pypi.python.org/pypi/async-class/
+   :alt: Wheel
+
+.. image:: https://img.shields.io/pypi/v/async-class
+   :target: https://pypi.org/project/async-class
+   :alt: Latest version
+
+
+async-class
+===========
+
+Adding abillity to write classes with awaitable initialization function.
+
+.. contents:: Table of contents
+
+Usage example
+=============
+
+.. code:: python
+
+
+   import asyncio
+   from async_class import AsyncClass, AsyncObject, task, link
+
+
+   class MyAsyncClass(AsyncClass):
+       async def __ainit__(self):
+           # Do async staff here
+           pass
+
+
+   class MainClass(AsyncObject):
+       async def __ainit__(self):
+           # Do async staff here
+           pass
+
+       async def __adel__(self):
+           """ This method will be called when object will be closed """
+           pass
+
+
+   class RelatedClass(AsyncObject):
+       async def __ainit__(self, parent: MainClass):
+           link(self, parent)
+
+
+   async def main():
+       instance = await MyAsyncClass()
+       print(instance)
+
+       main_instance = await MainClass()
+       related_instance = await RelatedClass(main_instance)
+
+       assert not main_instance.is_closed
+       assert not related_instance.is_closed
+
+       await main_instance.close()
+       assert not main_instance.is_closed
```

Comment:
why parent is not closed here if you wait close on the line above?

Reply:
Bug I guess